server/service/integration_core_test.go

```diff
@@ -7257,3 +7259,270 @@ func (s *integrationTestSuite) TestDirectIngestScheduledQueryStats() {
 	require.Equal(t, strconv.FormatInt(int64(sqs.UserTime), 10), row["user_time"])
 	require.Equal(t, strconv.FormatInt(int64(sqs.WallTime), 10), row["wall_time"])
 }
+
+// TestDirectIngestSoftwareWithLongFields tests that software with reported long fields
+// are inserted properly and subsequent reports of the same software do not generate new
+// entries in the `software` table. (It mainly tests the comparison between the currenly
+// inserted software and the incoming software from a host.)
+func (s *integrationTestSuite) TestDirectIngestSoftwareWithLongFields() {
```

Comment:
That's a great test, not sure if it should be in the integration suite though? AFAICS it only uses the datastore, which can be created for a test if needed, and doesn't require the end-to-end setup. Not a big deal, just mentioning in case you feel the same - I'm fine approving with the tests here.

Reply:
Yeah, placed it here because it tests how the direct ingest functions of `osquery_utils.GetDetailQueries` integrate with the real (MySQL) datastore:
```go
detailQueries = osquery_utils.GetDetailQueries(context.Background(), config.FleetConfig{}, appConfig, &appConfig.Features)
[...]
err = detailQueries["software_windows"].DirectIngestFunc(
  context.Background(),
  logger2,
  globalHost,
  s.ds,
  rows,
)
require.NoError(t, err)
```

Reply:
For that reason I'm leaving it here, let me know if that's a-ok.

Reply:
That's fine, my point was that you can create tests that use a real mysql datastore without adding to the integrations test suite, using `mysql.CreateMysqlDS` (https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/testing_utils.go#L268). We use that in various packages (not just inside the `datastore/mysql` package). But as I mentioned, not a big deal, I'm fine with leaving it there.